metrics/prometheus.go

```diff
@@ -0,0 +1,155 @@
+package metrics
+
+import (
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "traefik"
+)
+
+// TODO: #156 - More detailed metrics
+type entrypointMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type providerMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type frontendMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type backendMetrics struct {
+}
+
+type globalMetrics struct {
+	UpTime                    prometheus.Gauge
+	RequestResponseTimeTotal  prometheus.Gauge
```

Comment:
This is a counter. If you find yourself wanting to use Set on a counter then you should be using ConstMetrics and writing a custom collector rather than using direct instrumentation.